src/app/loadout-builder/processWorker/processUtils.ts

```diff
@@ -158,3 +157,162 @@ export function canTakeAllGeneralMods(
   // This will indicate we have iterated over all the mods, it will overshoot the length on success.
   return processedMods.length === modIndex;
 }
+
+/**
+ * This is heaps algorithm implemented for generating mod permutations.
+ * https://en.wikipedia.org/wiki/Heap%27s_algorithm
+ *
+ * Note that we ensure the array length is always 5 so mods are aligned
+ * with the 5 items.
+ */
+export function generateModPermutations(mods: ProcessMod[]): (ProcessMod | null)[][] {
+  const cursorArray = [0, 0, 0, 0, 0];
+  const modsCopy: (ProcessMod | null)[] = [...mods];
+
+  while (modsCopy.length < 5) {
+    modsCopy.push(null);
+  }
+
+  let i = 0;
+
+  const rtn = [Array.from(modsCopy)];
```

Comment:
You use `[...mods]` and `Array.from(mods)` sort of interchangeably

Reply:
Swapped the spread to an Array.from